docs/specs/component.md

```diff
@@ -9,24 +9,24 @@ interpreted as described in [RFC 2119].
 
 <!-- MarkdownTOC autolink="true" style="ordered" indent="   " -->
 
-1. [Introduction](#introduction)
-1. [Scope](#scope)
-1. [How to read this document](#how-to-read-this-document)
-1. [Naming](#naming)
-   1. [Source and sink naming](#source-and-sink-naming)
-   1. [Transform naming](#transform-naming)
-1. [Configuration](#configuration)
-   1. [Options](#options)
-      1. [`endpoint(s)`](#endpoints)
-1. [Instrumentation](#instrumentation)
-   1. [Batching](#batching)
-   1. [Events](#events)
-      1. [BytesReceived](#bytesreceived)
-      1. [EventsReceived](#eventsrecevied)
-      1. [EventsSent](#eventssent)
-      1. [BytesSent](#bytessent)
-      1. [Error](#error)
-1. [Health checks](#health-checks)
+- [Introduction](#introduction)
+- [Scope](#scope)
+- [How to read this document](#how-to-read-this-document)
+- [Naming](#naming)
+  - [Source and sink naming](#source-and-sink-naming)
+  - [Transform naming](#transform-naming)
+- [Configuration](#configuration)
+  - [Options](#options)
+    - [`endpoint(s)`](#endpoints)
```

Comment:
It wasn't changed here, but since it caught my eye: do we really want this to be a MUST in a component's configuration? Like, would we consider this for something like.... the S3 sink, or GCS sink, where the endpoint is well-known based on a combination of the region, bucket name, etc?

Reply:
Anyway, I agree, but I don't want to this to block getting this merged, so resolving for now.